go/worker/common/p2p/dispatch.go

```diff
@@ -165,6 +165,10 @@ func (h *topicHandler) dispatchMessage(peerID core.PeerID, m *queuedMsg, isIniti
 	defer h.handlersLock.RUnlock()
 
 	for _, handler := range h.handlers {
+		// Skip authentication if we are the peer.
```

Comment:
Would probably make more sense to just skip the for loop instead of breaking on the first iteration (as you don't need the `handler` for this check)?